GitCommands/Git/GitModule.cs

```diff
@@ -1514,7 +1514,8 @@ private ArgumentString GetFetchArgs(string remote, string remoteBranch, string l
                 { fetchTags == true, "--tags" },
                 { fetchTags == false, "--no-tags" },
                 { isUnshallow, "--unshallow" },
-                { prune, "--prune" }
+                { prune, "--prune" },
+                { prune && pruneTags, "+refs/tags/*:refs/tags/*".Quote() },
```

Comment:
1. There is a `--prune-tags` option. Why didn't you used it instead? To support older versions of git?

2. When I see the [documentation](https://git-scm.com/docs/git-fetch#_pruning) and your code, I understand that prune must me checked and also "prune tags" to really prune tags.
So I think that it is better to put some "radiobutton" logic in the UI (ie. when you check one, that uncheck the other one). So that the user understand that both can't be used at the same time. Instead of requiring by a tooltip that the "prune remote branches" checkbox must be checked otherwise the pruning of tags won't be done.

And in that case, that won't prune the remote branches, so it is misleading...

Reply:
> There is a --prune-tags option. Why didn't you used it instead? To support older versions of git?

"--prune-tags" option work only for "fetch" command not for "pull".
https://git-scm.com/docs/git-pull#Documentation/git-pull.txt---prune (No option.)

![image](https://user-images.githubusercontent.com/3169265/92221385-1014c580-eea6-11ea-9474-4c982708d966.png)


> When I see the documentation and your code, I understand that prune must me checked and also "prune tags" to really prune tags.
So I think that it is better to put some "radiobutton" logic in the UI (ie. when you check one, that uncheck the other one). So that the user understand that both can't be used at the same time. Instead of requiring by a tooltip that the "prune remote branches" checkbox must be checked otherwise the pruning of tags won't be done.

![image](https://user-images.githubusercontent.com/3169265/92220949-5fa6c180-eea5-11ea-9f7b-1d0915cf6270.png)
![image](https://user-images.githubusercontent.com/3169265/92220973-69c8c000-eea5-11ea-8646-7a790d66eecb.png)
![image](https://user-images.githubusercontent.com/3169265/92220995-73eabe80-eea5-11ea-813c-8ad0ece46616.png)
![image](https://user-images.githubusercontent.com/3169265/92221015-7d742680-eea5-11ea-8103-d484ee1a611f.png)

If the first checkbox is not selected, then the second is inactive. If you deselect the first, the second checkbox is deselected and deactivated

Make both checkboxes active?

Reply:
>Make both checkboxes active?

I think it's better, yes. And when you select one, it deselect the other one.
If I understand well, you replicated the command line behavior but it makes little sense in the UI because in the UI it is called "Prune remote branches" and not only "--prune" like in the command line.

Am I clear enough? Do you think it's better?

Reply:
In this case, it is better to make them independent and we will form a command according to the choice

Reply:
Although we cannot remove tags separately from branches

Reply:
Ок

Reply:
>In this case, it is better to make them independent and we will form a command according to the choice

I don't think so because if I understand well (but correct me if I'm wrong), if you prune tags, it won't prune the remote branches at the same time.
So if "Prune remote branches" is selected but the branches are not pruned, that's a bit misleading for the user, no?

Reply:
For "fetch" we can use "--prune --prune-tags",  works for branches and for tags.

```
$ git fetch origin --prune --prune-tags --verbose
 - [deleted]             (none)     -> origin/fix-6307
 - [deleted]             (none)     -> test-tag
 = [up to date]          fix-8430               -> origin/fix-8430
 ***
```
![image](https://user-images.githubusercontent.com/3169265/92235143-e6ff2f80-eebb-11ea-9c31-a87b1ee9fea7.png)


But for "pull" we nedd to use refspec "+refs/tags/*:refs/tags/\*". Works only for tags.

```
$ git pull origin --prune '+refs/tags/*:refs/tags/*' --verbose

 - [deleted]             (none)     -> test-tag
 = [up to date]          0.90           -> 0.90
 ***
```

![image](https://user-images.githubusercontent.com/3169265/92235334-2c236180-eebc-11ea-9e23-a05936ceec0a.png)

For some reason inactive for rebase.

![image](https://user-images.githubusercontent.com/3169265/92235430-54ab5b80-eebc-11ea-8e55-7af55b3ac558.png)

This option will also not work with '+refs/tags/*:refs/tags/*'.

![image](https://user-images.githubusercontent.com/3169265/92235755-e74bfa80-eebc-11ea-9d28-8033d0bb94cd.png)

I think we need to use this option only for "fetch" or split "pull" onto "fetch" and "merge"/"rebase".

Reply:
>I think we need to use this option only for "fetch" or split "pull" onto "fetch" and "merge"/"rebase".

I also think it makes sense to only allow this feature when doing a fetch.

>For some reason inactive for rebase.

I don't know why this option is available when pulling... I would disable it when doing a pull merge or rebase...

And also, if this `--prune-tags` could also be added to the "fetch and prune all" which is in the split button of the toolbar.